src/libguac/string.c

```diff
@@ -81,6 +81,17 @@ size_t guac_strlcat(char* restrict dest, const char* restrict src, size_t n) {
 
 }
 
+char* guac_strdup(const char* str) {
```

Comment:
Looks good, but we should add a unit test for this.

Reply:
Okay, wrote a couple of basic tests - not sure if there are other cases I should test in there??

Reply:
The tests you've added look good to me.